doc: openthread: add CSL feature information (#5501)
nrfconnect/sdk-nrf: doc/nrf/ug_thread_supported_features.rst

* Due to code size limitation, the combination of complete set of Thread 1.2 features with the Bluetooth® LE multiprotocol support is not possible for the nRF52833 DKs.

Coordinated Sampled Listening (CSL)
===================================

Coordinated Sampled Listening defined in IEEE 802.15.4-2015 is introduced by Thread 1.2 Specification to provide low latency communication for Sleepy End Devices.

> Thread 1.2 are required to support synchronized CSL transmissions to children which require them, so called Synchronized Sleepy End Devices (SSEDs).

Reviewer: ```suggestion
Thread 1.2 devices are required to support synchronized CSL transmissions to children which require them, known as Synchronized Sleepy End Devices (SSEDs).
```
Is devices the correct word here? Not 100% sure.

SSEDs are allowed to transmit frames normally at any time, but the routers should use the CSL transmission mechanism as long as the synchronization is maintained.

> This way an SSED is able to stay in a sleepy state more than 99% of the time and only turn on its radio periodically for a few hundreds of microseconds in order to receive frames from its parent.

Reviewer: ```suggestion
This allows an SSED to stay in a sleepy state more than 99% of the time and only turn on its radio periodically for a few hundreds of microseconds in order to receive frames from its parent.
```

> The most common use case for SSEDs is for developing low latency battery powered actuators, like window blinds.

Reviewer: ```suggestion
The most common use case for SSEDs is for developing low latency battery-powered actuators, such as window blinds.
```
